### PR TITLE
Added system tray icon

### PR DESCRIPTION
**Pull request by Novack**
Added TrayController class:
• Handles system tray icon and menu.
• Includes option for minimizing to tray (on/off).
• Includes red dot drawer for unread emails (corresponding events and unread counter on C# side not yet updated, as there are no messages in place from webview - placeholder menu items will help for testing this).
• Handles tooltip with unread mail counter.

Note: Added the existing logo image to the resources for using on the TrayController.

### CODE REVIEW

**Comment by Novack** [comment]
We need the messaging from skiff for informing and keeping updated the unread counter. Whenever that be in place, Im more than happy to use it on the unread counter and systry icon red dot.

In the same way, if we can have some messaging and RPCs for mail actions and info (ie: mark as read, delete, mark as spam, contact picture) I can do some work on adding some custom actions and enrichment to the toast notifications.

Edit: if you guys are busy, I will peek into the other projects to check if that is open source **and** if potentially I can give a hand with that.

**Comment by amilich** [comment]
Super exciting! Will review + test today.

**Comment by amilich** [comment]
Here's the code to send to Windows app - will be released today:
```
   const sendUnreadToWindowsApp = () => {
      if (!isWindowsDesktopApp()) {
        return;
      }
      try {
        const message = {
          type: 'unreadMailCount',
          data: { numUnread }
        };
        // Windows WebView2 injects a global `window.chrome` object
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
        chrome.webview.postMessage(JSON.stringify(message));
      } catch (error) {
        console.error('Failed to send data to Windows app', error);
      }
    };

    // Communicate with react-native and update about unread mail count (for updating iOS badge and macOS badge)
    // If we are on mobile app, send unread count to react-native
    sendRNWebviewMsg('unreadMailCount', { numUnread });
    // If we are on Windows app, send unread count to Windows app
    sendUnreadToWindowsApp();

    const interval = setInterval(() => {
      sendRNWebviewMsg('unreadMailCount', { numUnread });
      sendUnreadToWindowsApp();
    }, POLL_INTERVAL_IN_MS);
    return () => {
      clearInterval(interval);
    };
  }, [numUnread]);
```

**Comment by Novack** [comment]
This is awesome! Will check again later today then and add the necessary bindings. Much appreciated!

**Comment by Novack** [comment]
Implemented the new `unreadMailCount` message, and took the liberty to sort that message routing code into its own class in preparation for future messages. If the intention is to keep everything in the Windows class, let me know and I will change it back, (I would recomment following my propossed route).

You can check it out at https://github.com/skiff-org/skiff-windows-app/commit/b3e04dde144729dbbbd0da79f75480c95f4ce97c

Note that is not yet working, the message is not reaching the C# side yet.

**Comment by Novack** [comment]
Worth noting that notifications dont seem to be working, but looks like its not related to these changes, because its failing in my installed release as well. The internal flow processes the notification, but is never displayed. It looks like a lot of moving parts are involved in the toast messages, so not sure whats going on, probably worth simplifying a bit :)

Just in case, tomorrow will move to a previous commit and make sure I didnt broke anything.

Edit: looks like the issue is that the notifications wont display if the window is minimized, so not sure what was the intent.

Edit 2: it seems it was because I had the release 1.0.0.2. With the latest I do see the notifications, so will have to review whats going on. Will check later today.

**Comment by amilich** [comment]
> 

Checking on latest code here.

**Comment by amilich** [comment]
> Worth noting that notifications dont seem to be working, but looks like its not related to these changes, because its failing in my installed release as well. The internal flow processes the notification, but is never displayed. It looks like a lot of moving parts are involved in the toast messages, so not sure whats going on, probably worth simplifying a bit :)
> 
> Just in case, tomorrow will move to a previous commit and make sure I didnt broke anything.
> 
> Edit: looks like the issue is that the notifications wont display if the window is minimized, so not sure what was the intent.
> 
> Edit 2: it seems it was because I had the release 1.0.0.2. With the latest I do see the notifications, so will have to review whats going on. Will check later today.

Notifications should be fixed in latest push. Had to move the display to bottom right of screen instead of window.

**Comment by Novack** [comment]
Ok good news is, the unread counter code is working great, the tray icon is updated properly. I also sent a fix for an issue with the icon and removed some code used for demoing the unread dot, which is not required anymore. And sent a commit with the Skiff Desktop.csproj including the changes that are producing the conflict.

Still having issue with the notifications, but still not merged from your last cs yet.

**Comment by Novack** [comment]
Ok after merging from your last CS, I can confirm notifications are working as expected, with the window minimized and even with the window minimized to system tray.

There is another issue, where skiff announces a new email, but a few seconds pass until the mail (and the notification) actually appears in the list. This is an issue I have been experiencing from before all this, but is kind of more noticeable as the red dot also knows about the new email but is not yet there to see.

**Comment by amilich** [comment]
> Ok after merging from your last CS, I can confirm notifications are working as expected, with the window minimized and even with the window minimized to system tray.
> 
> There is another issue, where skiff announces a new email, but a few seconds pass until the mail actually appears in the list. This is an issue I have been experiencing from before all this, but is kind of more noticeable as the red dot also knows about the new email but is not yet there to see.

Yep, we'll have to fix this. The issue is that unread count polling is faster than entire inbox polling.

**Comment by Novack** [comment]
Gotcha. Just for reference, the internal skiff UI counter and the messages to native interfaces like C# work faster, while the actual listing and the notification happen afterwards.

**Comment by amilich** [comment]
> Gotcha. Just for reference, the internal skiff UI counter and the messages to native interfaces like C# work faster, while the actual listing and the notification happen afterwards.

Makes sense. That's consistent with not polling the actual messages more quickly.

**Review by amilich — APPROVED** [review]
looks great to me. tested and got the dot in my menubar, even when minimized, and notifications. will get a build out in the next day!

**Comment by Novack** [comment]
Awesome! Thank you for such a great service.

File: skiffWindowsApp/Skiff Desktop/TrayController.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            Brush bgColor = new SolidBrush(niceRed);
+            Bitmap bitmap = Properties.Resources.logo; // bitmap.Width: 32
+            Graphics graphics = Graphics.FromImage(bitmap);
+            int diameter = 14;
+            int coord = 18;
+            Rectangle dotPosAndSize = new Rectangle(coord, coord, diameter, diameter);
+            graphics.FillEllipse(bgColor, dotPosAndSize);
+            // If at some point we want to render the counter, we will need this, but will be redraw each time, so wont go here.
+            //graphics.DrawString(_count.ToString(), new Font("Arial Unicode MS", 11), Brushes.White, new PointF(coord, coord));
+            _iconUnreadDot = Icon.FromHandle(bitmap.GetHicon());
```
Comment (riverar) [inline skiffWindowsApp/Skiff Desktop/TrayController.cs:44]: `iconHandle` needs to be passed into `DestroyIcon`
`_iconNormal`should be disposed at some point
`graphics` needs to be disposed
`bitmap` needs to be disposed
`_iconUnreadDot` should be disposed at some point

Consider perhaps using a [SafeHandle](https://learn.microsoft.com/dotnet/api/system.runtime.interopservices.safehandle?view=net-6.0) implementation to handle icon lifetime for you, or explicitly implement a Dispose pattern.

**Review by riverar — COMMENTED** [review]
Added some nits on the icon resource handling code.

Opinion: I'm not entirely sold on the idea of a notification dot appearing in the system tray, for a productivity app. It should remain focused on alerting the user if there are more actionable items, such as new mail, etc.

**Pull request merged**